rfcs/2021-01-05-typography-scaling.md

```diff
@@ -0,0 +1,223 @@
+
+### **Typefaces**
+**Base:** We will be using [**Lato**](https://fonts.google.com/specimen/Lato?sidebar.open=true&selection.family=Lato:wght@400;700) as our base font family having two weights, **Regular(400)** & **Bold(700)**.
+
+**Code:** For code modules/snippets we'll use [**Menlo**](https://en.wikipedia.org/wiki/Menlo_(typeface)) font family throughout the system having two weights, **Regular** & **Bold**.
+
+**Fallbacks:** We'll use default system fonts (according to platform) as a fallback,
+```
+“Lato”, -apple-system, BlinkMacSystemFont, San Francisco, Segoe UI, Roboto, Helvetica Neue, sans-serif
+```
+
+### **Base font size**
+After auditing the whole product it came out that we are mostly using `13px` & `14px` as a base font-size on the body text. For the internal products, it is better to go with **`14px`** as a base font-size. And for small screens (mobile) it is better to go with **`15px`** as a base font-size. Because for dashboards the content is pretty dense and we need to maintain a small font-size ratio compared to marketing/landing pages. This will help the users to consume more content into a readable format. Although, we can still play with spacings so that it doesn't feel much complex.
+
+### **Line Height**
+Line heights also plays a significant role in the whole typography spectrum. If the font-size is smaller the text will be cramped and difficult to read, instead if it is larger the eyes can get lost. According to [WCAG](https://www.w3.org/WAI/WCAG21/Understanding/text-spacing.html), the line-height should be between `150% - 200%` of the font-size. But instead of blindly copying these values, we should also consider the font-face and platform on which this is getting rendered. Therefore, for optimal readability & accessibility aim for about `125% - 160%` line-height.
+
+For example,
+```
+.headingXXL-Desktop {
+  line-height: 1.25; /* looks good on large screens */
+  font-size: 40px;
+}
+
+.headingXXL-Mobile {
+  line-height: 1.5; /* looks good on small screens */
+  font-size: 28px;
+}
+```
+
+General Formula,
+
+>**`line-height = font-size * height-factor`**
+
+And will round-off the resulting value.
+
+### **Storage of Units**
+
+Units such as `line-height`, `letter-spacing`, `case` should not be stored separately. As the implementation doesn't have any logic to it. It is too much dependent on the type of font or the font-family we are using. So I'd suggest that we should store everything inside the typography scale itself.
```

Comment:
let's store the lineheight as a global token since we are advocating to use a common font across DS